cmd/gitannex/gitannex.go

```diff
@@ -267,6 +268,18 @@ func (s *server) handleInitRemote() error {
 		return fmt.Errorf("failed to get configs: %w", err)
 	}
 
+	var remoteExists bool
```

Comment:
I totally see your point about matching the behavior of other rclone commands, but there are a few reasons I think we should keep the current behavior. Let me know what you think.

1. The `rcloneremotename` parameter explicitly asks the user for a "remote name" as opposed to something more generic like "a remote". This intentionally matches the behavior of [git-annex-remote-rclone](https://github.com/git-annex-remote-rclone/git-annex-remote-rclone) to ease migration. (The `rcloneremotename` parameter is aliased to `target` for compatibility.)  https://github.com/rclone/rclone/blob/e334366345440af8c8392d47e662a8bc3607a4a4/cmd/gitannex/gitannex.go#L297-L298

2. If `rcloneremotename` supported paths in addition to a remote name, this would be the second way of specifying the remote path since we already have  `rcloneprefix`. This is not an insurmountable issue, but we would need to define the interaction between the two, i.e. what happens when a user says `rcloneremotename=foo:bar` *and* `rcloneprefix=baz`.

3.  I think we're already painted into a corner regarding backwards compatibility. If we start interpreting `rcloneremotename` as an *fs string* rather than a *remote name*, users will need to recreate/reconfigure their git-annex remotes.
\
   Today, if a user sets `rcloneremotename=foo`, we treat it as a remote name and internally build fs strings around it.  (And users are certainly not setting `remoteclonename=foo:` for a remote named `foo` — I believe [this line](https://github.com/rclone/rclone/blob/e334366345440af8c8392d47e662a8bc3607a4a4/cmd/gitannex/gitannex.go#L270) would construct an fs string that matches a remote named `foo:`, not `foo`.)
\
   If we started interpreting `rcloneremotename` as an fs string, users who have already set `rcloneremotename=foo` would be broken, right? Wouldn't that require us to interpret `foo` as a file path?

Reply:
Ah, I missed this in your other comment:

> Can we change that? We need to change it in a backwards compatible way so if you detect a string with no \/: in then stick a : on the end, otherwise assume it is a full rclone remote.

I think this neatly works around the issue I described in bullet 3. I'll have to mull on it a little. One edge case that immediately springs to mind is remotes that contain a colon in their name, but maybe that's a non-issue if it's rejected by the config parser.

Thanks!

Reply:
I didn't realise we had `rcloneremotename` and `rcloneprefix` - that works for me. Apologies!

It is a little non standard but lets run with it for this change and table the "changing the remote name in a backwards compatible way" discussion and get this working with the fstests!

I think what you should be doing then is using `fspath.SplitFs` on the fstest provided remote name

```
func SplitFs(remote string) (remoteName string, remotePath string, err error)
    SplitFs splits a remote a remoteName and an remotePath.

    SplitFs("remote:path/to/file") -> ("remote:", "path/to/file")
    SplitFs("/to/file") -> ("", "/to/file")

    If it returns remoteName as "" then remotePath is a local path

    The returned values have the property that remoteName + remotePath == remote
    (except under Windows where \ will be translated into /)
```

This then splits them into  `rcloneremotename` and `rcloneprefix`  for you.

I suspect you are going to want to munge the local directory case slightly - an obvious thing to do here would be to replace `""` with `":local:"`

Then with that hopefully most of the manipulations of the fstest remote can disappear and we can be sure that the tests only write in the directory they are given.

Does that sound like it will work?

Reply:
Great, glad we're on the same page!

I wasn't sure how to use the `:local:` remote -- it didn't seem to be the name of a remote? If there's some predefined local remote, that would let me get rid of the part that installs a custom config.

Reply:
Aha, thank you! This let me delete the wacky config installer part of the test code.

Reply:
Spoke too soon. It looks like the tests are creating a temp directory in the git repo under `cmd/gitannex/`. WIP.

Reply:
OK, I've got the special case for the local backend worked out now. 

I couldn't figure out how to get rid of the part in `handleInitRemote()` that checks whether the remote name has an entry in the config — I've got an explanatory comment over there and a new test case named "HandlesPrepareWithNonexistentRemote" that exercises the problematic scenario where the remote name is a local file path.